variants/sensebox_eye/APOTA.ino

```diff
@@ -0,0 +1,301 @@
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+  0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00
+};
+
+void IRAM_ATTR handleButtonPress() {
+  unsigned long currentTime = millis();  // Get current time
+
+  // Debounce: If the current press is too close to the last one, ignore it
+  if (currentTime - lastPressTime > 50) {
+    pressCount++;  // Count the button press
+
+    // Check if this is the second press within the double-press interval
+    if (pressCount == 2 && (currentTime - lastPressTime <= doublePressInterval)) {
+      doublePressDetected = true;  // Double press detected
+      pressCount = 0;              // Reset counter
+    }
+
+    lastPressTime = currentTime;  // Update the time of the last press
+  }
+}
+
+// Function to switch the boot partition to OTA1
+void setBootPartitionToOTA0() {
+  const esp_partition_t *ota0_partition = esp_partition_find_first(ESP_PARTITION_TYPE_APP, ESP_PARTITION_SUBTYPE_APP_OTA_0, NULL);
+
+  if (ota0_partition) {
+    // Set OTA1 as new boot partition
+    esp_ota_set_boot_partition(ota0_partition);
+    Serial.println("Boot partition changed to OTA0. Restarting...");
+
+    // Restart to boot from the new partition
+    esp_restart();
+  } else {
+    Serial.println("OTA1 partition not found!");
+  }
+}
+
+void setupDisplay() {
+  Wire.begin(PIN_QWIIC_SDA,PIN_QWIIC_SCL);
+  displayEnabled = Wire.requestFrom(0x3D, 1);  // Check if the display is connected
+  if (displayEnabled) {
+    display.begin(SSD1306_SWITCHCAPVCC, 0x3D);
+    display.display();
+    delay(100);
+    display.clearDisplay();
+  }
+}
+
+void displayStatusBar(int progress) {
+  display.clearDisplay();
+  display.setCursor(24, 8);
+  display.println("Sketch wird");
+  display.setCursor(22, 22);
+  display.println("hochgeladen!");
+
+  display.fillRect(0, SCREEN_HEIGHT - 24, SCREEN_WIDTH - 4, 8, BLACK);  // Clear status bar area
+  display.drawRect(0, SCREEN_HEIGHT - 24, SCREEN_WIDTH - 4, 8, WHITE);  // Draw border
+  int filledWidth = (progress * SCREEN_WIDTH - 4) / 100;                // Calculate progress width
+  display.fillRect(1, SCREEN_HEIGHT - 23, filledWidth - 4, 6, WHITE);   // Fill progress bar
+
+  display.setCursor((SCREEN_WIDTH / 2) - 12, SCREEN_HEIGHT - 10);
+  display.setTextSize(1);
+  display.setTextColor(WHITE, BLACK);
+  display.print(progress);
+  display.println(" %");
+  display.display();
+}
+
+void displayWelcomeScreen() {
+  display.clearDisplay();
+
+  // Draw WiFi symbol
+  display.drawBitmap(0, 12, epd_bitmap_wifi, 44, 44, WHITE);
+
+  // Display SSID text
+  display.setCursor(40, 13);
+  display.setTextSize(1);
+  display.setTextColor(WHITE, BLACK);
+  display.println("Verbinde dich");  // "Connect"
+  display.setCursor(60, 27);
+  display.println("mit:");  // "with"
```

Comment:
All log messages are English, why not also use English on the display?

Reply:
The board is mainly sold in germany and OTA is mainly used in workshops in schools also in germany

Reply:
I hope these students also learn to use English, as the software world is very much English oriented 😆

Reply:
At least it is not consistent. It is a mix of English and German.

Reply:
Don't they also look at the serial log output of the ESP?

Reply:
usually not. They flash it by drag-and-dropping bin files or over-the-air. The serial monitor is something that is very rarely used

Reply:
ok, I talked with a colleague and we are gonna switch it to english. I will update it in the pull request

Reply:
As I tried to suggest: English has taken over the IT world 🤓